client/ayon_deadline/repository/custom/plugins/GlobalJobPreLoad.py

```diff
@@ -651,6 +574,142 @@ def inject_ayon_environment(deadlinePlugin):
         raise
 
 
+def _wait_for_in_progress(export_url):
+    """Check if another worker started extractenvironments already.
+
+    extractenvironments might be expensive operation, so first worker who
+    starts doing it creates empty file before it. All other workers should
+    wait until this file gets renamed to final name.
+    Raises:
+        (RuntimeError) if extraction takes more
+            than EXTRACT_ENVIRONMENT_TIMEOUT seconds
+    """
+    start_time = datetime.now()
+    while os.path.exists(f"{export_url}.tmp"):
+        date_diff = datetime.now() - start_time
+        if date_diff > timedelta(seconds=EXTRACT_ENVIRONMENT_TIMEOUT):
+            raise RuntimeError(
+                "Previous extract environment process stuck for "
+                f"'{EXTRACT_ENVIRONMENT_TIMEOUT}' sec."
+                "Starting it from scratch."
+            )
+        print("Extract environment process already triggered, waiting")
+        sleep(2)
+
+
+def _get_output_dir(job):
+    """Look for output dir where metadata.json should be created also."""
+    output_urls = ["OutputFilePath", "Output", "SceneFile"]
+    output_dir = None
+    for output in output_urls:
+        output_dir = job.GetJobPluginInfoKeyValue(output)
+        if output_dir:
+            break
+    if not output_dir:
+        raise RuntimeError(
+            "Unable to find workfile location or "
+            "location where files should be rendered.")
+    output_dir = os.path.dirname(output_dir)
+    return output_dir
+
+
+def _extractenvironments(
```

Comment:
This function should be `_extract_environments`. It lost new line characters during movement, please put them back, it is unreadable when it's squashed.